vnstat2: add package

janh opened the pull request:
**Maintainer:** me

**Compile tested:**
 - x86-64, KVM, master/19.07
 - ipq40xx, AVM FRITZ!Box 4040, 19.07

**Run tested:**
 - x86-64, KVM, master/19.07
   (tested in default configuration and with persistent database, import from vnStat 1.x works)
 - ipq40xx, AVM FRITZ!Box 4040, 19.07
   (running for 2 weeks now with imported data from vnStat 1.x)

**Description:**

This adds an additional package for vnStat 2.x which uses an SQLite database instead of struct dumps and supports a shorter 5-minute resolution as well as longer data retention than vnStat 1.x.

The UCI configuration from previous versions is partially supported: Configured interfaces are added to the vnStat database on start/reload. Other options (symlink, backup, download) are not implemented.

hnyman commented on net/vnstat2/files/vnstat.init:
```diff
@@ -0,0 +1,60 @@
+#!/bin/sh /etc/rc.common
+
+START=99
```
Why 99?
One more service for the "race to be the last one"...

/etc/rc.local should be run last and is started as "95" (as S95done).
Please decrease this to start before that.

janh replied:
Thanks for the comment. I didn't really think about this value and just kept the one from the original vnstat package. I'll change it.

diizzyy replied:
As it's not a critical service and there are some race conditions I think 99 was chosen for that reason.

janh replied:
It looks like the `START` value of 99 for vnStat has been there since 2008: https://git.openwrt.org/?p=openwrt/svn-archive/archive.git;a=blob;f=net/vnstat/files/vnstat.init;h=07db20cba863daca068409a24f03affa12d9952a;hb=b9358dc52de6b8abd99f7b57e18cc68ab4a8b0f9

A race condition is mentioned in https://github.com/openwrt/packages/pull/4682 and https://github.com/openwrt/packages/pull/4615#issuecomment-321765674. Personally, I have never seen this issue (I'm using vnStat with database on a USB stick since 2015). Even `START=60` seems to work for now, but of course I don't have any long-term experience for that.

Another thing I could see is that the system time is more likely to be correct with `START=99` (sysntpd is started as 98). But there is no guarantee for that anyway and this is just a cosmetic issue. And if a persistent database is used vnStat deals with this by itself (by waiting for a configurable time after startup if the system time is before the latest database entry).

janh commented:
I changed the init script `START` value to 60 (this is still after fstab, so it is possible to store the database on an external device). I also added a `STOP` value so that vnStat is properly stopped and can save its database before the storage is unmounted.

hnyman commented:
@jow- 
Any opinions on this? 
Should this replace vnstat, or should vnstat 1.0 be left there as a separate package?

diizzyy commented:
It depends on how long we want to keep deprecated hardware around more or less as vnstat2 pulls in libsqlite which makes it quite a bit larger in the end compared to v1.x. 1.x is also is unsupported upstream.

neheb commented:
@hnyman https://github.com/openwrt/packages/pull/7676#issuecomment-448903210

diizzyy commented:
Any reason why autotools aren't used?
